portfolio/src/main/java/com/google/sps/servlets/GetCommentsServlet.java

```diff
@@ -41,44 +40,41 @@
   private static final String COMMENT = "Comment";
 
   /**
-   * On GET request, writes to the response the comments list as a JSON string.
+   * {@inheritDoc}
+   *
+   * Returns the most recently posted comments.
+   *
+   * This servlet is called every time comments.html is loaded. The number of comments to send is
+   * specified by {@code maxComments}.
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Obtain and prepare comments from Datastore.
     Query commentsQuery = new Query(COMMENT).addSort("timestamp", SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    PreparedQuery commentsPrepared = datastore.prepare(commentsQuery);
+    List<Entity> commentsPrepared =
+      datastore.prepare(commentsQuery).asList(FetchOptions.Builder.withLimit(maxComments));
 
-    // Loop through each Comment entity until all comments are seen or until the max number of
-    // comments have been reached, and store in an ArrayList.
+    // Loop through each Comment entity until all comments are seen, storing them in an ArrayList
```

Comment:
I think this comment is unnecessary given the simplification of the code in this change.

Reply:
Yeah I think it's clearer now that Ananya pointed out the withLimit functio I can use. I've deleted the comment.